File: Week4/VolvoCodingChallenge/app/src/main/java/com/yeah/ruisu/volvocodingchallenge/data/remote/RemoteService.java
package com.yeah.ruisu.volvocodingchallenge.data.remote;

import com.yeah.ruisu.volvocodingchallenge.data.remote.models.CityID;
import com.yeah.ruisu.volvocodingchallenge.data.remote.models.ForecastDatum;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface RemoteService
{
    @GET("api/location/search/")
    Single<CityID> getWOEID(@Query("query") String cityName);

    @GET("api/location/{WOEID}/{year}/{month}/{day}/")
    Single<List<ForecastDatum>> getWeatherForecast(@Path("WOEID") String WOEID,
                                                   @Path("year") String year,
                                                   @Path("month") String month,
                                                   @Path("day") String day);
}

File: week5/RealVolvoCodingChallenge/app/src/main/java/com/yeah/ruisu/realvolvocodingchallenge/data/remote/RemoteService.kt
package com.yeah.ruisu.realvolvocodingchallenge.data.remote

import com.yeah.ruisu.realvolvocodingchallenge.data.remote.models.ForecastDatum
import kotlinx.coroutines.experimental.Deferred
import retrofit2.http.GET
import retrofit2.http.Path

interface RemoteService
{
    @GET ("api/location/{WOEID}/{date}")
    fun getWeatherForecast(@Path("WOEID") WOEID:String,
                          @Path("date") date:String)
            : Deferred<List<ForecastDatum>>;
}